test/e2e_node/device_plugin.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
+	dp "k8s.io/kubernetes/pkg/kubelet/cm/deviceplugin"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Serial because the test restarts Kubelet
+var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin] [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("device-plugin-errors")
+
+	Context("DevicePlugin", func() {
+		By("Enabling support for Device Plugin")
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FeatureGates[string(features.DevicePlugins)] = true
+		})
+
+		BeforeEach(func() {
+
+		})
+
+		AfterEach(func() {
+
+		})
+
+		It("Checks that Kubelet handles device plugin registration correctly.", func() {
+
+			By("Start stub device plugin")
+			devs := []*pluginapi.Device{
+				{ID: "Dev-1", Health: pluginapi.Healthy},
+				{ID: "Dev-2", Health: pluginapi.Healthy},
+			}
+
+			dp1 := dp.NewDevicePluginStub(devs, pluginapi.DevicePluginPath+"dp.1")
+			err := dp1.Start()
+			framework.ExpectNoError(err)
+
+			By("Wait for node is ready")
+			Eventually(func() int {
+				nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+				return len(nodeList.Items)
+			}, 30*time.Second, framework.Poll).Should(Equal(1))
+
+			By("Register resources")
+			resourceName := "mock1.com/resource"
+			err = dp1.Register(pluginapi.KubeletSocket, resourceName)
+			framework.ExpectNoError(err)
+
+			By("Waiting for stub device plugin to become available on the local node")
```

Comment:
s/Waiting for stub device plugin/Waiting for the resource exported by the stub device plugin